### ISSUE TITLE
ElasticSearch: Support for v7 ? 

### ISSUE THREAD

**Issue by peternitro** [akka/alpakka#2025]
### Short description

It looks like the alpakka ES project is compiled against ES client v6.3.1. ES 7.x has been out for a while and is supported by AWS.  

### Details

Using a 7.x client is not possible (binary interface changes results in a runtime exception) and using a 6.x client with a 7.x server results in the following warnings:

```
WARN  o.e.c.RestClient - request [POST http://localhost:9200/_bulk] returned 1 warnings: [299 Elasticsearch-7.1.1-7a013de "[types removal] Specifying types in bulk requests is deprecated."]
```

**Comment by ennru** [comment]
Alpakka Elasticsearch uses the JSON interface to be able to talk to different versions of ES. As the ES JSON changes over time we'd need to introduce some version support as I suggested in https://github.com/akka/alpakka/issues/1232#issuecomment-534723623

**Comment by ennru** [comment]
https://www.elastic.co/guide/en/elasticsearch/reference/7.x/removal-of-types.html

**Comment by ennru** [comment]
We just merged #2052 so that we can implement different REST messages according to the ES version.
Would be great if ES users can help and implement support for ES 6 and ES 7.

**Comment by ennru** [comment]
Implemented with #2277
